waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -299,20 +305,13 @@ proc paginateWithIndex*(list: seq[IndexedWakuMessage], pinfo: PagingInfo): (seq[
       newCursor = msgList[s].index # the new cursor points to the begining of the page
 
   if (retrievedPageSize == 0):
-    return (@[], PagingInfo(pageSize: 0, cursor:pinfo.cursor, direction: pinfo.direction))
+    return (@[], PagingInfo(pageSize: 0, cursor:pinfo.cursor, direction: pinfo.direction), HistoryResponseError.OK)
```

Comment:
What is HistoryResponseError.OK? Seems like an oxymoron, isn't 0 pages retrieved, if correct, an OK response?

Reply:
The goal of the error message is to distinguish between two cases that both result in 0 message history. 1) when the cursor is invalid 2) the cursor is valid but there is no match for the submitted query
When an error message is OK, it means the second scenario happened. Having zero messages in the response is not by itself an error, but if it is because of an invalid cursor then it is an error (as the input is invalid).

Reply:
Following @jm-clius comment https://github.com/status-im/nim-waku/pull/660#discussion_r666993971, I have renamed OK to NONE to resolve the naming ambiguity.